### ISSUE TITLE
Intellisense not displaying TypeScript autocomplete.

### ISSUE THREAD

**Issue by nbering** [Microsoft/vscode#3889]
I upgraded to 0.10.10 this morning and immediately the TypeScript language support broke.

No squiggly hints for broken code, no Intellisense completion. I tried changing `intellisense.tsdk` along with a handful of typescript versions, but it didn't help.

Grasping at straws I set TSS_DEBUG and TSS_TRACE in my .bash_profile and all that got me was a notice that the TypeScript server had crashed. This suggests that VSCODE is indeed starting the TypeScript service and it is probably not failing (it worked in 0.10.9).

Downgrading fixed the issue for me for now, but I'd like to help diagnose the problem or find workarounds if possible. If someone could point me in the right direction for language service diagnostics I'd be happy to provide more feedback to help reproduce this issue.

OS: OS X 10.11.3 El Capitan
VSCode: 0.10.10
Typescript: Packaged with VSCode, 1.8.7, 1.9.0-dev.20160226

**Comment by egamma** [comment]
Thanks for your offer to help diagnose this further. 

To isolate the problem further can you configure the `tsdk` setting in the downgraded version of Code (that works for you) to point to the same TypeScript version that ships with VS Code 0.10.10, that is 1.8.2. Do you still experience the crash?

**Comment by dbaeumer** [comment]
@nbering would it be possible for you to share your workspace. Then we could try to reproduce this locally as well.

**Comment by nbering** [comment]
Alright, I did some extra exploration. I could not isolate the issue specifically, but it does seem to be somehow related to my project, though I'm not sure how exactly.

I pointed VSCode 0.10.9 at Typescript 1.8.2 and it worked fine. I upgraded 0.10.10, set to the same typescript version and it did not work.

However, 0.10.10 did work with another typescript project I had on my machine. Unfortunately, I cannot share the project that is not working as I am legally obligated to keep that information private.

I can give some details about the project, though and maybe you can help narrow this down.

The project is an Ionic/Cordova application built on their starter project. The project has `gulp` installed with `gulp-typescript`. External typescript references are being managed by `typings`. The typescript project uses internal modules with the `--outFile` option.

The project that is working also uses internal modules and `--outFile`, and is structured very similarly but just uses `npm` to manage tasks. It also uses `typings` to manage external typescript references. I tried setting everything but the file paths to the same in tsconfig.json and the broken project still does not work.

In summary: the issue seems to be project-specific but not an issue with typescript, because it works with the same version of typescript in the back-version of vscode.

**Comment by Cleod9** [comment]
Hi,

So I think it's possible I'm having an issue that is related to this. Seems that it affects TypeScript versions greater than 1.8.*. Basically VS Code fails for me on all `import` statements that use non-relative paths (i.e. paths that necessitate traversing up the file tree to locate a folder). As a result, I get "module not found" errors and my auto-completion breaks completely.

If I downgrade typescript in my node_modules to 1.7.5, and set the following in my settings.json the problem goes away:

`"typescript.tsdk": "node_modules/typescript/lib"`

I don't know for sure if it's the same issue, but if it's related hopefully it might provide a clue!

**Comment by dbaeumer** [comment]
@Cleod9 which module system are you using. Commonjs. In the past the tsc compiler was not correct in resolving modules in commonjs and they fixed this. In CommonJs there is no concept of a baseURL as in AMD. But the tsc compiler nevertheless walked up the parent chain. No if in commonjs and you want to reference another module you need to provide a relative path. It is the same as in JS with the require statement.

**Comment by Cleod9** [comment]
@dbaeumer Yes we are using the commonjs setting in `tsconfig`. Note that we also use ES6-style imports for all TS modules.

Do you have a recommendation for what to change our tsconfig to? Webpack is able to build just fine because we're using its `modulesDirectories` config option, but there doesn't seem to be a way to reference extra modules folders besides `node_modules` in the current build for VS Code (although it seems it is planned for future iirc?)

**Comment by nbering** [comment]
My particular case is using Internal Modules only, no imports and no commonjs modules. So unless there's something in my definitely typed ambient module definitions there's nothing related to module resolution. It's also specific to VSCode v0.10.10 because it works with the same version of typescript in the back-version of VSCode, but if your case helps to identify the conditions in which the VSCode Intellisense component stops taking input from the TypeScript language service I'm totally willing run diagnostics on test solutions on my project.

**Comment by nbering** [comment]
Explicitly adding the extension to my triple-slash references seems to have solved my problem.

```TypeScript
/// <reference path="App" />
//becomes
/// <reference path="App.ts" />
```

After discovering this, I was able to upgrade to v0.10.11.
